src/protocols/vnc/display.c

```diff
@@ -153,6 +153,7 @@ void guac_vnc_copyrect(rfbClient* client, int src_x, int src_y, int w, int h, in
 }
 
 void guac_vnc_set_pixel_format(rfbClient* client, int color_depth) {
+    client->format.trueColour = 1;
```

Comment:
Yes, that's correct. The default in rfbGetClient() in libvncserver which you're calling is already setting:
client->format.trueColour = TRUE;

This is just overriding that value so that it's 1 instead of -1.